datafusion/core/src/datasource/stream.rs

```diff
@@ -103,19 +105,46 @@ impl FromStr for StreamEncoding {
     }
 }
 
-/// The configuration for a [`StreamTable`]
+/// The StreamProvider trait is used as a generic interface for reading and writing from streaming
+/// data sources (such as FIFO, Websocket, Kafka, etc.).  Implementations of the provider are
+/// responsible for providing a `RecordBatchReader` and optionally a `RecordBatchWriter`.
+pub trait StreamProvider: std::fmt::Debug + Send + Sync {
```

Comment:
`StreamFormat` might be a better name here, to align with `FileFormat`

Reply:
I am not sure people associate location with 'Format'. FileFormat does not deal with location. I think StreamProvider is descriptive enough.

Reply:
i viewed location as specific to the `FileStreamProvider` and `Format` as a concept that provided reading and writing capabilities, similar to `FileFormat`.  But i dont feel strongly either way so im fine to leave at `StreamProvider`.